src/backend/core/ai/utils.py

```diff
@@ -0,0 +1,116 @@
+import uuid
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Any, Dict, List
+
+from core.models import Thread
+
+
+@dataclass
+class Message:
```

Comment:
It feels weird to create a second model here. If what you basically need is a message serializer it should be written like this, or even be integrated in the mail model